### PR TITLE
2130 ge behavior annex support

### LINKED ISSUE #2130: Visualization and editing of behavior annex state automata
Visualization and editing of behavior annex state automata, except for expressions.

The high-level state machine will be displayed and edited in the OSATE graphical editor alongside other AADL model elements. Behavior specifications will be represented by a behavior diagram element. Like other model elements, the user will be able to set the “in modes” clause and other attributes using the Eclipse Properties view. States, transitions, and variables will be represented inside the behavior specification element.

The label for behavior specification diagram element will be based on the containing classifier and the mode(s) to which they apply. If the diagram element is contained within a classifier, the label may be shortened to avoid redundancy. 

A Behavior Specification diagram type will be introduced which will show states and transitions by default. 

Additionally, adjust setup to stop hiding the `org.osate.ge.ba` project and start including it with OSATE.

Ensure copy and pasting works for BA elements.

### PR DESCRIPTION
**Pull request by RyanMcilnay**
Closes #2130

### CODE REVIEW

**Review by lwrage — CHANGES_REQUESTED** [review]
This PR adds files for org.alisa.common.tests, which was removed a while ago. These need to be removed from the PR.

File: core/org.osate.annexsupport/src/org/osate/annexsupport/AnnexValidator.java [hunk 1]
```diff
@@ -1,126 +1,126 @@
-package org.osate.annexsupport;
```
Comment (lwrage) [inline core/org.osate.annexsupport/src/org/osate/annexsupport/AnnexValidator.java:1]: What happened with this file?

**Review by lwrage — CHANGES_REQUESTED** [review]
Update copyright year in file headers to 2021.

File: core/org.osate.annexsupport/src/org/osate/annexsupport/AnnexValidator.java (hunk 1 above)
Comment (RyanMcilnay) [reply core/org.osate.annexsupport/src/org/osate/annexsupport/AnnexValidator.java]: Not really sure, it was showing changes although none had occurred.  Fixed it now though.

**Comment by RyanMcilnay** [comment]
> This PR adds files for org.alisa.common.tests, which was removed a while ago. These need to be removed from the PR.

Thank you for catching this.  Fixed now.

File: ge/org.osate.ge/src/org/osate/ge/ui/PropertySectionUtil.java [hunk 2]
```diff
@@ -96,4 +98,24 @@ public static Label createSectionLabel(final Composite container,
 		label.setLayoutData(fd);
 		return label;
 	}
+
+	/**
+	 * Creates a button with the specified data, selection listener, text, and type
```
Comment (philip-alldredge) [inline ge/org.osate.ge/src/org/osate/ge/ui/PropertySectionUtil.java:103]: Move this back to the internal class.

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/util/InternalPropertySectionUtil.java [hunk 3]
```diff
@@ -75,14 +74,6 @@ public static void setPropertiesHelp(final Control control) {
 		ContextHelpUtil.setHelp(control, ContextHelpUtil.PROPERTIES_VIEW);
 	}
 
-	public static Button createButton(final TabbedPropertySheetWidgetFactory widgetFactory, final Composite composite,
```
Comment (philip-alldredge) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/util/InternalPropertySectionUtil.java:78]: Restore this.

**Review by philip-alldredge — CHANGES_REQUESTED** [review]
I left some comments. I had a few other general comments:

**Comment by philip-alldredge** [comment]
@RyanMcilnay please see the comments and let me know if you have any questions. Be sure to look into the general comments that were left at the start of the review.

**Comment by RyanMcilnay** [comment]
@philip-alldredge I think createParsedAnnexSubclause in org.osate.ge.aadl2.GraphicalAnnexUtil is new and needs exposed.  Previously, there was only a getOrCreate. The createParsedAnnexSubclause method is required for BA specification that is created from palette.

**Comment by philip-alldredge** [comment]
Okay go ahead and expose it.

Philip Alldredge
Principal Research Scientist
Rotorcraft Systems Engineering and Simulation Center
University of Alabama in Huntsville
VBH A-4
256-824-4837
Philip.Alldredge@uah.edu

**Comment by philip-alldredge** [comment]
Closing. Another pull request has been created. Use it. https://github.com/osate/osate2/pull/2600

**Pull request closed without merging**